compiler/rustc_target/src/abi/mod.rs

```diff
@@ -1257,6 +1257,14 @@ impl<'a> LayoutS<'a> {
             align,
         }
     }
+
+    /// Compare two `LayoutS` while disregarding the Abi of either layout.
+    pub fn equal_up_to_abi<'b>(&self, other: &LayoutS<'b>) -> bool {
```

Comment:
Ugh, not pertaining to this PR, but this reminds me that we ended up with the `S` suffix anti-pattern (the suffix should be more like `Data` or something less jarring, and that type be avoided to be named - the lack of `Deref` is really troubling, as that is how you avoid forcing getter methods everywhere).

Not sure what this PR can do, not seeing any methods on `Layout`, hmpf.